File: examples/python/minimal/snippets/datacapture/document_data_extractor.py
from scanbotsdk import *
from utils import print_generic_document

def extract_document_data(image: ImageRef):
    # Make sure you have a valid license
    license_info = get_license_info()
    if license_info.status != LicenseStatus.OKAY:
        return
    
    config = DocumentDataExtractorConfiguration()
    config.result_accumulation_config.confirmation_method = ConfirmationMethod.EXACT
    # Configure other parameters as needed.
    
    extractor = DocumentDataExtractor(configuration=config)
    result: DocumentDataExtractionResult = extractor.run(image=image)

    # if you want to use cropped image later, call result.cropped_image.encode(), and save buffer,
    # otherwise image reference will be released when the ImageRef object is garbage collected.
    
    print(f"Extraction Status: {result.status.name}")
    print_generic_document(result.document)

File: examples/nodejs/src/snippets/datacapture/document-data-extractor.ts
import * as ScanbotSDK from "scanbotsdk";
import { printGenericDocument } from "../utils/utils";

export class DocumentDataExtractorSnippet {
  public static async run(image: ScanbotSDK.ImageRef): Promise<void> {
    var commonConfig =
      new ScanbotSDK.DocumentDataExtractorCommonConfiguration();
    commonConfig.acceptedDocumentTypes = [
      "DeIdCardFront",
      "DeIdCardBack",
      "DeHealthInsuranceCardFront",
      "DePassport",
      "DeResidencePermitFront",
      "DeResidencePermitBack",
      "EuropeanHealthInsuranceCard",
      "EuropeanDriverLicenseFront",
      "EuropeanDriverLicenseBack",
    ];
    // Configure supported document types
    const elements: ScanbotSDK.DocumentDataExtractorConfigurationElement[] = [
      commonConfig,
    ];

    const accumulationConfig =
      new ScanbotSDK.ResultAccumulationConfiguration();
    accumulationConfig.confirmationMethod = "EXACT";

    const config = new ScanbotSDK.DocumentDataExtractorConfiguration();
    config.configurations = elements;
    config.resultAccumulationConfig = accumulationConfig;
    // Configure other parameters as needed.

    // `await using` ensures both extractor and result are properly disposed
    // when the scope ends, as they hold unmanaged resources.
    await using extractor = await ScanbotSDK.DocumentDataExtractor.create(config);
    await using result = await extractor.run(image);

    this.printResult(result);
  }

  private static printResult(
    result: ScanbotSDK.DocumentDataExtractionResult
  ): void {
    console.log(`Status: #${result.status}`);
    printGenericDocument(result.document);
  }
}
